subsys/disk/disk_access.c

```diff
@@ -124,47 +124,52 @@ int disk_access_register(struct disk_info *disk)
 {
 	int rc = 0;
 
-	k_mutex_lock(&mutex, K_FOREVER);
+	k_spinlock_key_t spinlock_key = k_spin_lock(&lock);
 	if ((disk == NULL) || (disk->name == NULL)) {
 		LOG_ERR("invalid disk interface!!");
 		rc = -EINVAL;
 		goto reg_err;
 	}
+	k_spin_unlock(&lock, spinlock_key);
```

Comment:
Looks like there is no need for a lock and you can replace `goto` with `return -EINVAL`, and you can remove rc variable. Only sys_dlist_append() needs to be protected, same applies for the  disk_access_unregister().